packages/components/src/custom-select-control/index.js

```diff
@@ -81,6 +101,13 @@ export default function CustomSelectControl( {
 		stateReducer,
 	} );
 
+	const anchorRef = useRef();
+
+	const [
+		buttonResizeListener,
+		{ width: buttonWidth },
```

Comment:
Note to self: this currently sets the width relative to the _button_ width, which is technically working nicely, but I _think_ on trunk, the width is set relative to the _parent_. This difference results in the following:

| Storybook on trunk | Storybook on this PR |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/146321921-be7958b4-7a11-4a9f-af62-07fded18c749.png) | ![image](https://user-images.githubusercontent.com/14988353/146321942-f0e9c0a5-2fd0-4ceb-8de0-4f6f32b269cd.png) |

Moving this up a level to get the width of the parent is probably the next step, we then might need to switch the position of the Popover to use an `anchorRect` or a `getAnchorRect` instead of `anchorRef` to set the right position relative to the width. I'll have a play tomorrow.

Reply:
Resolved, I moved the `anchorRef` and width calculation up to the container in https://github.com/WordPress/gutenberg/pull/37272/commits/21c2f016349ad3c65284e15a1399782063f18e0f which solved most things.